lib/elixir/lib/task.ex

```diff
@@ -255,4 +267,131 @@ defmodule Task do
   def find(_tasks, _msg) do
     nil
   end
+
+  @doc """
+  Yields, temporarily, for a task reply.
+
+  Returns `{:ok, reply}` if the reply is received.
+
+  A timeout, in milliseconds, can be given with default value
+  of `5000`. In case of the timeout, this function will return `nil`
+  and the monitor will remain active. Therefore `yield/2` can be
+  called multiple times on the same task.
+
+  In case the task process dies, this function will exit with the
+  same reason as the task.
+
+  This function assumes the task's monitor is still active or the monitor's
+  `:DOWN` message is in the message queue. If it has been demonitored, or the
+  message already received, this function wait for the duration of the timeout
+  awaiting the message.
+  """
+  @spec yield(t, timeout) :: {:ok, term} | nil
```

Comment:
Ah, I just found out that `Task.find/2` returns `nil`. So I would keep yield returning nil but I am still more inclined to make shutdown return just ok.